### ISSUE TITLE
Why is my SPMC crashing 

### ISSUE THREAD

**Issue by rhinoTV** [koying/SPMC#815]
Nvidia Shield TV, crashing on open.  Here are the logs:

paste.semperpax.com/pslt3g1zr

**Comment by JFG90** [comment]
Why are you not on the latest version!? No crashes

**Comment by Afullmark** [comment]
> Why are you not on the latest version!? No crashes

@JonnyGadd The latest version is not release in the google play store yet, which might be why.

**Comment by JFG90** [comment]
Then side load it? I never install the playstore version it's always best to download direct from the official website

**Comment by rhinoTV** [comment]
Thanks guys. 

I was actually still on 16.4.2 when crashes and strange behavior started. It was working great for me up to them, so didn't update. 16.5.5 had trouble even starting. I'm on 16.6 now and it is still crashing after a short time. It's probably not SPMC but I think it's a skin add-on like Skin Helper Service but need help interpreting the logs. 

I'll post a new log file in a few minutes if anyone can help and check it out.

**Comment by war59312** [comment]
16.6 crashes on my Shield 2015 like every other time starting up:

http://paste.semperpax.com/p26th45ud

@rhinoTV, mind retitling this issue to something like: Crashing On Run (Nvidia Shield)

**Comment by koying** [comment]
Looks like an addon trying to retrieve a non-existing window property.
I cannot tell the addon nor the property...

**Comment by rhinoTV** [comment]
My SPMC configuration had been working great on the Shield for a long time. I figured the crashes may have been due to a corrupted file or an add-on that was updated. I thought the best solution was to delete my addon and userdata folders and copy over my backups from January.

I made the backups just before I adopted a 128GB USB flash drive. I spent many hours struggling trying to delete and replace the files via FTP, Mac, Es File Explorer, and Xplorer. I kept getting errors and messages like " xxx cannot be deleted". FTP would delete some files then error.

If I powered off the shield, removed the USB, powered it back up, powered down again and reinserted the USB, and powered on , I could sometimes make changes, but many files just would not delete. So I couldn't replace them. Very frustrating. Contacted Nvidia, but they were no help. 

I tried uninstalling SPMC thinking maybe the files were locked or something. Didn't work. Then I reinstalled SPMC from the Play store, it would open with the default Confluence skin. But... when I restart the Shield SPMC is no longer found in my apps. Google play store also shows it is not installed.

Basically spent all day trying to fix it and just made things worse. Seems like it is a big problem with the USB flash drive. It's like it loses the connection or permissions with it. Perhaps overheated? Now I'm thinking I will have to delete all of my game files so I can migrate back to the smaller internal flash storage and hopefully get SPMC working again. 

This is the USB flash drive I am using (SanDisk SDCZ43-128G-GAM46) and it is one of the ones Nvidia recommends:

https://www.amazon.com/gp/product/B01BGTG2A0

@koying or anyone, have you seen this issue before? It's there a better way to fix it?

**Comment by SaxonRix** [comment]
@rhinoTV hmmm, I seem to be having a similar problem. It doesn't crash at start up, but probably 10-15 minutes into running SPMC. 

I have two Nvidia Shield TV's (Both 2015) and the crashing ONLY happens on the one which happens to have a USB stick in it (SanDisk Extreme 64 GB USB, formatted as internal storage). This has literally only just started happening (it was fine yesterday!), so I think only few things could be the culprit:

- Potentially a dying USB stick (I've had it just under a year, always plugged into my Shield).
- A corrupt or poorly updated add-on (Although I'm not sure why it would still only be happening on one of my Shields if that was the case).

I know this most likely isn't anything to do with SPMC directly, but it would be great if we could keep this issue open, so we can try and get to the bottom of it.

**Comment by SaxonRix** [comment]
@rhinoTV I've now moved all the data from the adoptable storage USB onto the internal storage, re-formatted (as regular storage) and removed the USB stick and rebooted. SPMC now seems to work great again, no crashes. I didn't lose any settings either.

A bit lame, but luckily for me I didn't really need the extra storage space anyway as I only really use the shield for streaming apps.

Maybe try that, and potentially try re-applying the USB adoptable storage again in-case it was just the formatting that became corrupt or something.

**Comment by rhinoTV** [comment]
@SaxonRix That's exactly what I ended up doing. I moved enough files off of the USB flash drive so I could migrate back to the internal storage. There were still files I couldn't delete, or rename which made it difficult to replace my SPMC data. But I was able to move the folders of these files to a folder I named "undeletable". Now not sure I want to try applying the drive again. I may RMA it back to SanDisk for replacement. Wish I knew what caused the files to be corrupt, the drive, or FTP, or the Shield.

**Comment by SaxonRix** [comment]
@rhinoTV @koying I've started to have this issue again. I've not re-insterted the USB, so I know its not that now. I have uploaded a log of the latest crash here, if it helps? http://paste.semperpax.com/p3eayuzfu

I tried to upload the log from within SPMC too, but was getting errors.

The issue seems to happen even if i just load up SPMC and then leave it idle for 5-10 minutes. It's quite frustrating because it only happens on one of my Shields and they both have the same setup...

Any help would be appreciated. Let me know if there's any more info you might need in order to diagnose the problem.
